files/Aboot/boot0.j2

```diff
@@ -392,7 +392,8 @@ extract_image() {
 
 extract_image_secureboot() {
     info "Extracting necessary swi content"
-    unzip -oq "$swipath" platform/firsttime .imagehash -d "$image_path"
+    # NOTE: boot/ is not used by the boot process but only extracted for kdump
```

Comment:
>boot/ is not used by the boot process but only extracted for kdump [](start = 12, length = 66)

Why this is made specially for secureboot? How about the normal boot?

Reply:
In secureboot we don't want to reuse the extracted kernel on the flash because it could have been tampered with.
Instead we extract kernel/initrd from the SWI which is verified at boot time and kexec these which removes this attack vector.

Reply:
Is it true that normal build extracts this 'boot/*'?

---
In reply to: [562265581](https://github.com/Azure/sonic-buildimage/pull/6447#discussion_r562265581) [](ancestors = 562265581)

Reply:
Yes they do.
Now secureboot do to. Not doing so would make `kdump-tools` report an error during boot after a crash has occured.
Hence the reason I now exctract kernel and initrd. Technically the kernel would be enough, but there's no harm in extracting both.